modules/howtos/examples/analytics.cc

```diff
@@ -0,0 +1,161 @@
+                                    code->valueint, msg->valuestring);
+                            }
+                        }
+                    }
+                }
+            }
+            delete[] data;
+        }
+    }
+
+    if (lcb_respanalytics_is_final(resp)) {
+        std::cout << "META: ";
+        printf("\x1b[1mMETA:\x1b[0m ");
+    } else {
+        std::cout << (*idx)++ << " ";
+    }
+    std::cout << std::string(row, nrow) << "\n";
+    if (lcb_respanalytics_is_final(resp)) {
+        std::cout << "\n";
+    }
+
+    lcb_DEFERRED_HANDLE *handle = nullptr;
+    lcb_respanalytics_deferred_handle_extract(resp, &handle);
+    if (handle) {
+        const char *status;
+        size_t status_len;
+        lcb_deferred_handle_status(handle, &status, &status_len);
+        std::cout << "DEFERRED: " << std::string(status, status_len) << "\n";
+        lcb_deferred_handle_callback(handle, row_callback);
+        check(lcb_deferred_handle_poll(instance, idx, handle), "poll deferred query status");
+        lcb_deferred_handle_destroy(handle);
+    }
+}
+
+int main(int argc, char *argv[])
+{
+    lcb_INSTANCE *instance;
+
+    if (argc < 2) {
+        std::cerr << "Usage: " << argv[0] << " couchbase://host/beer-sample [ password [ username ] ]\n";
+        exit(EXIT_FAILURE);
+    }
+
+    {
+        lcb_CREATEOPTS *create_options = nullptr;
+        lcb_createopts_create(&create_options, LCB_TYPE_BUCKET);
+        lcb_createopts_connstr(create_options, argv[1], strlen(argv[1]));
+        if (argc > 3) {
+            lcb_createopts_credentials(create_options, argv[3], strlen(argv[3]), argv[2], strlen(argv[2]));
+        }
+        check(lcb_create(&instance, create_options), "create couchbase handle");
+        lcb_createopts_destroy(create_options);
+        check(lcb_connect(instance), "schedule connection");
+        lcb_wait(instance, LCB_WAIT_DEFAULT);
+        check(lcb_get_bootstrap_status(instance), "bootstrap from cluster");
+        {
+            char *bucket = nullptr;
+            check(lcb_cntl(instance, LCB_CNTL_GET, LCB_CNTL_BUCKETNAME, &bucket), "get bucket name");
+            if (strcmp(bucket, "beer-sample") != 0) {
+                fail(LCB_ERR_INVALID_ARGUMENT, "expected bucket to be \"beer-sample\"");
+            }
+        }
+    }
+
+    {
+        // tag::analytics[]
+        const char *stmt = "SELECT * FROM breweries LIMIT 2";
+        lcb_CMDANALYTICS *cmd;
+        int idx = 0;
+        lcb_cmdanalytics_create(&cmd);
+        lcb_cmdanalytics_callback(cmd, row_callback);
+        lcb_cmdanalytics_statement(cmd, stmt, strlen(stmt));
+        lcb_cmdanalytics_deferred(cmd, 1);
+        check(lcb_analytics(instance, &idx, cmd), "schedule analytics query");
+        std::cout << "----> " << stmt << "\n";
+        lcb_cmdanalytics_destroy(cmd);
+        lcb_wait(instance, LCB_WAIT_DEFAULT);
+    }
+    // end::analytics[]
```

Comment:
Minor thing, we can probably move the antora `end` tag just below line 155. Just so it doesn't look odd in the code block.

Reply:
fixed